LeaderboardBackend.Test/Fixtures/PostgresDatabaseFixture.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Threading.Tasks;
+using DotNet.Testcontainers.Builders;
+using DotNet.Testcontainers.Configurations;
+using DotNet.Testcontainers.Containers;
+using LeaderboardBackend.Test.Fixtures;
+using Npgsql;
+using NUnit.Framework;
+
+// Fixtures apply to all tests in its namespace
+// It has no namespace on purpose, so that the fixture applies to all tests in this assembly
+
+[SetUpFixture] // https://docs.nunit.org/articles/nunit/writing-tests/attributes/setupfixture.html
+internal class PostgresDatabaseFixture
+{
+	public static TestcontainerDatabase? PostgresContainer { get; private set; }
+	public static bool HasCreatedTemplate { get; private set; } = false;
+
+	private static string TemplateDatabase => PostgresContainer?.Database + "_template";
+
+	[OneTimeSetUp]
+	public static async Task OneTimeSetup()
+	{
+		if (TestConfig.DatabaseBackend != DatabaseBackend.TestContainer)
+		{
+			return;
+		}
+
+		PostgresContainer = new ContainerBuilder<PostgreSqlTestcontainer>()
+			.WithDatabase(new PostgreSqlTestcontainerConfiguration
+			{
+				Database = "db",
+				Username = "dbuser",
+				Password = "weft5gst768sr",
+			})
+			.WithTmpfsMount("/var/lib/postgresql/data") // db files in-memory
+			.Build();
+		await PostgresContainer.StartAsync();
+	}
+
+	[OneTimeTearDown]
+	public static async Task OneTimeTearDown()
+	{
+		if (PostgresContainer is null)
+		{
+			return;
+		}
+
+		await PostgresContainer.DisposeAsync();
+	}
+
+	public static void CreateTemplateFromCurrentDb()
+	{
+		ThrowIfNotInitialized();
+
+		NpgsqlConnection.ClearAllPools(); // can't drop a DB if connections remain open
+		using NpgsqlDataSource conn = CreateConnectionToTemplate();
+		conn.CreateCommand(@$"
+			DROP DATABASE IF EXISTS {TemplateDatabase};
+			CREATE DATABASE {TemplateDatabase}
+				WITH TEMPLATE {PostgresContainer!.Database}
+				OWNER '{PostgresContainer!.Username}';
+			")
+			.ExecuteNonQuery();
```

Comment:
Sorry but I'm having a hard time following this. So `CreateConnectionToTemplate()` returns a DataSource that's connected to `PostgresContainer`, but with its database set to "template1". But `conn.CreateCommand()` would create a database named "db_template", wouldn't it? If so, do we use "template1" anywhere?